### PR TITLE
Cleaned up libraries and changes the high level modules include statements

### PR DESCRIPTION
**Pull request by Chases5**
This changes our header files to use pragma once so we are no longer including common.h multiple times. This also changes common.h to be Waypoint.h and most the actual common items to the new common.h. All the deprecated functions in the new Waypoint.h have been removed including very old elcano serial stuff.

### CODE REVIEW

**Comment by ptressel** [comment]
Chase --

> This also changes common.h to be Waypoint.h and most the actual common items to the new common.h.

Do you mean that most of what was in common.h is now in Waypoint.h?  Did you use git mv to make Waypoint.h, and then edit it?  That might be useful to retain the history of common.h in Waypoint.h, if most of it is the same.  Not necessary though.

**Comment by muddmaker** [comment]
Do we want to also take this opportunity to add namespaces and use a consistent styling convention?

**Comment by ptressel** [comment]
@muddmaker  Not in the same PR.  It's good to split up different types of large non-functional (cleanup) changes.  Makes it easier to undo.  Plus, we can get them approved more quickly if they're split, as we might want to argue about one little piece, and that would stall the rest.  :-D

**Comment by ptressel** [comment]
@Chase5 Do you know for certain that the Arduino compiler (meaning, the official one used by the Arduino IDE) knows about pragma once?  That is not a standard -- it's just something that some compilers support.  If you are not certain and have not tested, please replace this by the standard #ifdef guards.

**Comment by TylerFolsom** [comment]
In FusionData.h, HeadingPrecision and compass_error should be changed to give their units, e.g. compass_error_deg.
Should the remaining statements in Common.h go to Settings.h?

File: libraries/Kalman/Matrix.h [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#pragma once
+#ifndef MATRIC
```
Comment (ptressel) [inline libraries/Kalman/Matrix.h]: Typo?

File: libraries/FusionData/FusionData.h [hunk 2]
```diff
@@ -1,12 +1,13 @@
-#pragma once
+#ifndef FUSIONDATA
```
Comment (ptressel) [inline libraries/FusionData/FusionData.h]: Tradition uses something like FUSIONDATA_H.  That way it doesn't look like some normal constant.

File: libraries/FusionData/FusionData.h [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-
-#define HEADING_PRECISION 1000
+#define HEADING_PRECISION_DEG 1000
 // Hard coded values
-#define COMPASS_ERROR ((double) 2)
-#define DISTANCE_ERROR ((double) 0.04)
-#define GPS_ERROR ((long) 4236)
+#define COMPASS_ERROR_DEG ((double) 2)
+#define DISTANCE_ERROR_MM ((double) 0.04)
+#define GPS_ERROR_DEG ((long) 4236)
```
Comment (ptressel) [inline libraries/FusionData/FusionData.h]: Unless these constants are not used anywhere else, it would be better to make name changes in a separate PR, and update the files that use them in the same PR.  (That's another Tradition:  He who makes the change, fixes the dependencies. :-D  A.k.a. Don't break the build!)

File: libraries/FusionData/FusionData.h [hunk 4]
```diff
@@ -1,12 +1,13 @@
+#ifndef FUSIONDATA
```
Comment (ptressel) [inline libraries/FusionData/FusionData.h]: By Tradition, header exclusion constants get _H on the end (i.e. FUSIONDATA_H here) so they don't accidentally collide with normal constants.

File: libraries/FusionData/FusionData.h [hunk 5]
```diff
@@ -1,7 +1,8 @@
...
```
Comment (ptressel) [inline libraries/FusionData/FusionData.h]: FUSIONDATA_H because Tradition and name collision avoidance.

File: libraries/GPS/GPS.h [hunk 6]
```diff
@@ -1,3 +1,5 @@
+#ifndef GPS
```
Comment (ptressel) [inline libraries/GPS/GPS.h]: GPS_H

File: libraries/Kalman/Matrix.h [hunk 7]
```diff
@@ -1,3 +1,5 @@
+#ifndef MATRIX
```
Comment (ptressel) [inline libraries/Kalman/Matrix.h]: MATRIX_H

File: libraries/Waypoint/Waypoint.h [hunk 8]
```diff
@@ -0,0 +1,95 @@
+#ifndef WAYPOINT
```
Comment (ptressel) [inline libraries/Waypoint/Waypoint.h]: WAYPOINT_H

**Comment by Chases5** [comment]
I need to push another PR ASAP so I'll branch this one and save it for later.
